pkg/deployment/ipam.go

```diff
@@ -208,24 +196,50 @@ func (dns *DataplaneDNSData) EnsureDNSData(ctx context.Context, helper *helper.H
 			dataplanev1.NodeSetDNSDataReadyCondition,
 			condition.ErrorReason, condition.SeverityError,
 			dataplanev1.NodeSetDNSDataReadyErrorMessage)
-		return err
+		return dnsDetails, err
 	}
-
 	if !dnsData.IsReady() {
 		util.LogForObject(helper, "DNSData not ready yet waiting", instance)
 		instance.Status.Conditions.MarkFalse(
 			dataplanev1.NodeSetDNSDataReadyCondition,
 			condition.RequestedReason, condition.SeverityInfo,
 			dataplanev1.NodeSetDNSDataReadyWaitingMessage)
-		dns.Ready = false
-		return nil
+		return dnsDetails, nil
 	}
+
 	instance.Status.Conditions.MarkTrue(
 		dataplanev1.NodeSetDNSDataReadyCondition,
 		dataplanev1.NodeSetDNSDataReadyMessage)
-	dns.Ready = true
+	dnsDetails.IsReady = true
+	return dnsDetails, nil
+}
 
-	return nil
+// EnsureIPSets Creates the IPSets
+func EnsureIPSets(ctx context.Context, helper *helper.Helper,
```

Comment:
Why are you moving this?

Reply:
It's arranging the DNS and IPSet functions be at one place. Not mix them like before.